core/src/test/java/nu/yona/server/subscriptions/service/PrivateUserDataMigrationServiceIntegrationTest.java

```diff
@@ -85,95 +95,75 @@ public void upgrade(User userEntity)
 	@MockBean
 	private MessageSourceRepository mockMessageSourceRepository;
 
-	@MockBean
-	private UserAnonymizedRepository mockUserAnonymizedRepository;
+	@Autowired
+	private UserAnonymizedRepository userAnonymizedRepository;
 
 	@Autowired
-	private UserService service;
+	private PrivateUserDataMigrationService service;
 
-	private final String password = "password";
-	private User john;
+	@Autowired
+	private UserService userService;
 
-	private List<Class<? extends MigrationStep>> originalMigrationSteps;
+	private final String password = "password";
+	private User richard;
 
-	@Before
-	public void setUpForAll()
+	@BeforeClass
+	public static void setUpForAll()
 	{
 		LocaleContextHolder.setLocale(Translator.EN_US_LOCALE);
-
-		originalMigrationSteps = PrivateUserDataMigrationService.getMigrationSteps();
-		PrivateUserDataMigrationService.setMigrationSteps(Arrays.asList(MockMigrationStep1.class, MockMigrationStep2.class));
-	}
-
-	@After
-	public void teardownForAll()
-	{
-		PrivateUserDataMigrationService.setMigrationSteps(originalMigrationSteps);
 	}
 
 	@Before
 	public void setUpPerTest()
 	{
+		userAnonymizedRepository.deleteAll();
 		JUnitUtil.setUpRepositoryMock(mockUserRepository);
```

Comment:
Why not convert those mocks too?

Reply:
My approach is that I use a Mockito mock wherever I can and only use our own mocks when Mockito is insufficient.